src/index.js

```diff
@@ -280,7 +280,10 @@ function getMain({ options, entry, format }) {
 		mainNoExtension,
 	);
 	mainsByFormat.modern = replaceName(
-		(pkg.syntax && pkg.syntax.esmodules) || pkg.esmodule || 'x.modern.js',
+		(typeof pkg.exports === 'string' && pkg.exports) ||
```

Comment:
This works for the main case. I'm wondering - maybe we could walk the exports object to find matching `import` entries?

```
function walk(exports) {
  if (typeof exports === 'string') return exports;
  return walk(exports['.'] || exports.import || exports.module);
}

walk(pkg.exports)
```

Reply:
Certainly could. I started to think about this but had reservations on whether or not users would expect all keys (or at least the ones in the spec?) to be supported. If they have a `"require"`, would they then expect a matching output file? Not sure.